File: 백준/Gold/1806. 부분합/부분합.cc
#include <bits/stdc++.h>
using namespace std;
int n, s, a;
vector<int> v;

int main() {
    cin >> n >> s;
    for(int i=0; i<n; i++) {
        cin >> a;
        v.push_back(a);
    }
    int left=0, right=0;
    int sum = 0;
    int minLen = INT_MAX;
    
    while(true) {
        if(sum >= s) {
            minLen = min(minLen, right - left);
            sum -= v[left++];
        } else if (right == n) { 
            break;
        } else {
            sum += v[right++];
        }
    }
    if(minLen == INT_MAX) {
        cout << 0;
    } else {
        cout << minLen;
    }
}

File: 백준/Gold/1806. 부분합/부분합.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for(int i=0; i<n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int minLen = Integer.MAX_VALUE;
        int sum = 0;
        int left = 0;
        for(int right=0; right<n; right++) {
            sum += arr[right];
            while(sum >= s) {
                minLen = Math.min(minLen, right-left+1);
                sum -= arr[left];
                left++;
            }
        }
        if(minLen == Integer.MAX_VALUE) {
            System.out.println(0);
        } else {
            System.out.println(minLen);
        }
    }
}